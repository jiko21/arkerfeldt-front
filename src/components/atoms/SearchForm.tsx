/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import { FC, InputHTMLAttributes } from 'react';
import { Color } from '@/const/color';
import { faSearch } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const wrapperStyle = css`
  height: 30px;
  position: relative;
  max-width: 240px;
`;

const svgStyle = css`
  position: absolute;
  top: 4px;
  right: 0;
  height: 24px;
  width: 24px;
`;

const inputStyle = css`
  border: 1px solid ${Color.BLACK};
  border-radius: 6px;
  padding: 0 0 0 2px;
  height: 100%;
  width: 100%;

  ::placeholder {
    color: ${Color.LIGHT};
  }
`;

interface IProps extends InputHTMLAttributes<HTMLInputElement> {
  testId?: string;
}

const SearchForm: FC<IProps> = ({ testId, ...props }) => (
  <div css={wrapperStyle}>
    <span css={svgStyle}>
      <FontAwesomeIcon
        icon={faSearch}
        css={css`
          color: ${Color.BLACK};
        `}
      />
    </span>
    <input
      css={inputStyle}
      type={'text'}
      placeholder={'search keyword'}
      {...props}
      data-testid={testId}
    />
  </div>
);

export default SearchForm;
