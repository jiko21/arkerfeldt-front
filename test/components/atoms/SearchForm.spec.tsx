import SearchForm from '@/components/atoms/SearchForm';
import { fireEvent, render, screen } from '@testing-library/react';

describe('SearchForm.tsx', () => {
  let onChangeMock: jest.Mock;
  beforeEach(() => {
    onChangeMock = jest.fn();
  });

  afterEach(() => {
    onChangeMock.mockClear();
  });
  it('should correctly render', () => {
    const { container } = render(
      <SearchForm value={'sample'} onChange={onChangeMock} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should correctly call onClick', () => {
    const TEST_ID = 'TEST_ID';
    render(
      <SearchForm value={'sample'} onChange={onChangeMock} testId={TEST_ID} />,
    );
    fireEvent.change(screen.getByTestId(TEST_ID), {
      target: {
        value: 'test',
      },
    });
    expect(onChangeMock).toBeCalled();
  });
});
