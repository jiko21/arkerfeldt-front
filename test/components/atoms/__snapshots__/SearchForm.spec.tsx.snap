// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm.tsx should correctly render 1`] = `
.emotion-0 {
  height: 30px;
  position: relative;
  max-width: 240px;
}

.emotion-1 {
  position: absolute;
  top: 4px;
  right: 0;
  height: 24px;
  width: 24px;
}

.emotion-2 {
  color: #2d2d2d;
}

.emotion-3 {
  border: 1px solid #2d2d2d;
  border-radius: 6px;
  padding: 0 0 0 2px;
  height: 100%;
  width: 100%;
}

.emotion-3::-webkit-input-placeholder {
  color: #878686;
}

.emotion-3::-moz-placeholder {
  color: #878686;
}

.emotion-3:-ms-input-placeholder {
  color: #878686;
}

.emotion-3::placeholder {
  color: #878686;
}

<div>
  <div
    class="emotion-0"
  >
    <span
      class="emotion-1"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-search fa-w-16 emotion-2"
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
        />
      </svg>
    </span>
    <input
      class="emotion-3"
      placeholder="search keyword"
      type="text"
      value="sample"
    />
  </div>
</div>
`;
